addon/types/index.ts

```diff
@@ -1,14 +1,15 @@
 export type WaiterName = string;
+export type Token = unknown;
 
 export interface IWaiter {
   name: WaiterName;
   waitUntil(): boolean;
   debugInfo(): unknown;
 }
 
-export interface ITestWaiter<T> extends IWaiter {
-  beginAsync(item: T, label?: string): void;
-  endAsync(item: T): void;
+export interface ITestWaiter extends IWaiter {
```

Comment:
Why remove the generic? It still seems accurate, and removing it means that you _can't_ enforce types for the token if you want to...

Reply:
You're right, `T` should be the generic type for token.

Reply:
@scalvert and I discussed this, and I feel pretty strongly that `unknown` is appropriate for the type of the token, given that we allow consumers of the library to hold onto it, but don't want them to depend on its specific value.

Enforcing types for the token is possible _through type guards_, and which is appropriate in my opinion b/c it's only meaningful at runtime.

Reply:
We've managed to come up with a happy medium, where we default `T` to `Token`.